//
// Bentobox
// Proto
// Value
//

syntax = "proto3";
package bento.protos;

import "bento/protos/types.proto";

message Value {
  // Data type held by this value
  Type data_type = 1;

  // Defines a value of the primitive kind
  message Primitive {
    oneof value {
      // primitive values
      int32 int_8 = 1;
      int32 int_32 = 2;
      int64 int_64 = 3;
      float float_32 = 4;
      double float_64 = 5;
      string str_val = 6;
      bool boolean = 7;
    }
  }
  // Defines a value of the array kind
  message Array {
    repeated Primitive values = 1;
  }

  oneof kind {
    Primitive primitive = 2;
    Array array = 3;
  }
}
